src/nvim/event/socket.c

```diff
@@ -40,17 +40,18 @@ int socket_watcher_init(Loop *loop, SocketWatcher *watcher,
       return UV_EINVAL;
     }
 
-    *host_end = '\0';
     uv_getaddrinfo_t request;
+    *host_end = '\0';
+    char *host = !strcmp(addr, "localhost") ? "127.0.0.1" : addr;
```

Comment:
Why are we jumping through all these hoops for the theoretical case of someone specifying `localhost` when that isn't a known hostname on their system?  What's the goal here?